t/unit-tests/u-example-decorate.c

```diff
@@ -0,0 +1,74 @@
+#define USE_THE_REPOSITORY_VARIABLE
+
+#include "unit-test.h"
+#include "object.h"
+#include "decorate.h"
+#include "repository.h"
+
+struct test_vars {
+	struct object *one, *two, *three;
+	struct decoration n;
+	int decoration_a, decoration_b;
+};
+
+static struct test_vars vars;
+
+void test_example_decorate__add(void)
+{
+	void *ret = add_decoration(&vars.n, vars.one, &vars.decoration_a);
+	cl_assert(ret == NULL);
+	ret = add_decoration(&vars.n, vars.two, NULL);
+	cl_assert(ret == NULL);
+}
+
+void test_example_decorate__readd(void)
+{
+	void *ret = add_decoration(&vars.n, vars.one, &vars.decoration_a);
+	ret = add_decoration(&vars.n, vars.one, NULL);
+	cl_assert(ret == &vars.decoration_a);
+	ret = add_decoration(&vars.n, vars.two, NULL);
+	cl_assert(ret == NULL);
+	ret = add_decoration(&vars.n, vars.two, &vars.decoration_b);
+	cl_assert(ret == NULL);
```

Comment:
This test here still intermingles the setup with the actual logic you want to test. Something like this would help:

```

void test_example_decorate__readd(void)
{
	void *ret

	cl_assert(add_decoration(&vars.n, vars.one, &vars.decoration_a) == NULL);
	cl_assert(add_decoration(&vars.n, vars.two, NULL) == NULL);

	ret = add_decoration(&vars.n, vars.one, NULL);
	cl_assert(ret == &vars.decoration_a);
	ret = add_decoration(&vars.n, vars.two, &vars.decoration_b);
	cl_assert(ret == NULL);
}
```